### PR TITLE
document new `AT TIME ZONE` operators

### LINKED ISSUE #5481: docs: document new `AT TIME ZONE` operators
We got a bunch of great new support for timezone-related stuff in v0.6.1, but it didn't come with docs! This issue tracks writing documentation for that feature.

The PR is https://github.com/MaterializeInc/materialize/pull/5130.

We'll also want to retroactively add it to the release notes for v0.6.1. There's a placeholder in there now. https://github.com/MaterializeInc/materialize/blame/b172eae9633687c67f2bb1c8c570b3b50635bfe3/doc/user/content/release-notes.md#L179

@quodlibetor do you think you could quickly sketch out what was added? Just a list of the functions/new syntax and links to their docs in PostgreSQL would be great!

### PR DESCRIPTION
**Pull request by msudberg**
Fixes #5481 

This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6069)

### CODE REVIEW

**Review by quodlibetor — APPROVED** [review]

**Comment by sploiselle** [comment]
@msudberg Could you provide more detail about what these PRs are rather than just the issue number in the title? You can e.g. link to the issue by prefixing the issue ID number with a hash sign, e.g. `#6069`.

File: doc/user/content/release-notes.md
Comment (sploiselle) [inline doc/user/content/release-notes.md:255]: "conversion of" is either passive voice or feels like it; using verbs makes writing feel much less plodding and monolithic, e.g. "Supporting converting `timestamp`...."

**Comment by msudberg** [comment]
I can do that for the future. I didn't think it was necessary because the
PR is linked to the originating ticket.

File: doc/user/content/sql/functions/timezone-and-at-time-zone.md
Comment (sploiselle) [inline doc/user/content/sql/functions/timezone-and-at-time-zone.md:29]: `timestamp with time zone` is usually just written as `timestamptz` in the docs

File: doc/user/sql-grammar/sql-grammar.bnf
Comment (sploiselle) [inline doc/user/sql-grammar/sql-grammar.bnf:279]: non-keyword elements should not be placed in quotation marks

**Review by sploiselle — CHANGES_REQUESTED** [review]
LGTM modulo the railroad diagrams which have to be fixed before merging

File: doc/user/content/release-notes.md
Comment (msudberg) [reply doc/user/content/release-notes.md:255]: "Conversion of timestamp" is the object phrase, not the verb; the verb is "Support", which is in the active voice. I think "Support converting" or "Handle converting" sound awkward, so I'm leaving as-is.

Comment (sploiselle) [reply doc/user/content/release-notes.md:255]: Ah, I typoed my suggestion, which should have been "Support converting `timestamp`...". You'll notice this is the format of most other release notes. I believe you've used the argument that consistent formatting is of paramount importance, so not sure why that doesn't apply here?

**Pull request merged**
